probe/endpoint/conntrack.go

```diff
@@ -143,15 +164,45 @@ func (c *Conntracker) run(args ...string) {
 		return
 	}
 
+	defer func() {
+		log.Printf("contrack exiting")
+	}()
+
 	// Now loop on the output stream
 	decoder := xml.NewDecoder(reader)
 	for {
 		var f Flow
 		if err := decoder.Decode(&f); err != nil {
 			log.Printf("conntrack error: %v", err)
+			return
+		}
+		c.handleFlow(f, false)
+	}
+}
+
+func (c *Conntracker) existingConnections(args ...string) ([]Flow, error) {
+	var conntrack conntrack
+	args = append([]string{"-L", "-o", "xml", "-p", "tcp"}, args...)
+	cmd := exec.Command("conntrack", args...)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return conntrack.Flows, err
```

Comment:
If you return non-nil err, best to return `[]Flow{}`, to prevent ambiguity in the caller.